Mathlib/Algebra/Order/Module/Defs.lean

```diff
@@ -806,6 +806,52 @@ instance instSMulPosReflectLE [SMulPosReflectLE α β] : SMulPosReflectLE α β
 end Right
 end OrderDual
 
+section StrictOrderedSemiring
+variable [StrictOrderedSemiring α] [OrderedAddCommMonoid β] [Module α β] [ExistsAddOfLE α]
+  [ExistsAddOfLE β]
+
+section PosSMulMono
+variable [PosSMulMono α β] [ContravariantClass β β (· + ·) (· ≤ ·)] {a₁ a₂ : α} {b₁ b₂ : β}
+
+/-- Binary **rearrangement inequality**. -/
+lemma smul_add_smul_le_smul_add_smul (ha : a₁ ≤ a₂) (hb : b₁ ≤ b₂) :
+    a₁ • b₂ + a₂ • b₁ ≤ a₁ • b₁ + a₂ • b₂ := by
```

Comment:
... this inequality is still true if you replace `β` by `βᵒᵈ`, but its assumptions are too strong to prove this. Indeed, `ExistsAddOfLE β` does not imply `ExistsAddOfLE βᵒᵈ`

We would need a weaker typeclass which would imply this inequality and would be preserved under `ᵒᵈ`. `ExistsAddOrAddOfLE` in my TODO is one possible typeclass.